model/command_webhook.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package model
+
+type CommandWebhook struct {
+	Id        string
+	CreateAt  int64
+	CommandId string
+	UserId    string
+	ChannelId string
+	RootId    string
+	ParentId  string
+	UseCount  int
+}
+
+const (
+	COMMAND_WEBHOOK_LIFETIME = 1000 * 60 * 30
+)
+
+func (o *CommandWebhook) PreSave() {
+	if o.Id == "" {
+		o.Id = NewId()
+	}
+
+	if o.CreateAt == 0 {
+		o.CreateAt = GetMillis()
+	}
+}
+
+func (o *CommandWebhook) IsValid() *AppError {
+	if len(o.Id) != 26 {
+		return NewLocAppError("CommandWebhook.IsValid", "model.command_hook.id.app_error", nil, "")
```

Comment:
Can you use `NewAppError` here with the `http.StatusBadRequest`? If the error propagates all the way back to the HTTP response it's good to have this set to non-500